include/swift/AST/DiagnosticsFrontend.def

```diff
@@ -618,5 +618,15 @@ ERROR(ast_format_requires_dump_ast,none,
 ERROR(unknown_dump_ast_format,none,
       "unknown format '%0' requested with '-dump-ast-format'", (StringRef))
 
+ERROR(dependency_scan_unexpected_variant, none,
```

Comment:
Can we downgrade this from error to a warning? I suspect this would happen semi-frequently that a fatal error is probably an overkill.

Reply:
We should discuss more about the implication with @cyndyishida and other folks. Please hold off merging before that.